app/helpers/pieces_helper.rb

```diff
@@ -0,0 +1,14 @@
+module PiecesHelper
+  def castling(destination_piece)
```

Comment:
I was planning to extract the logic for castling from the pieces controller in this module, and then call it from the pieces controller .

 What do you think?